pkg/controller/sync/accessor.go

```diff
@@ -216,62 +219,71 @@ func (a *resourceAccessor) FederatedResource(eventSource util.QualifiedName) (Fe
 			// Ensure the federated name is namespace qualified.
 			federatedName.Namespace = federatedName.Name
 		} else {
+			if federatedName.Namespace != federatedName.Name {
+				// A FederatedNamespace is only valid for propagation
+				// if it has the same name as the containing namespace.
+				return nil, nil
+			}
 			// Ensure the target name is not namespace qualified.
 			targetName.Namespace = ""
 		}
+	}
+
+	key := federatedName.String()
+
+	resource, err := util.ObjFromCache(a.federatedStore, kind, key)
+	if err != nil {
+		return nil, err
+	}
+	if resource == nil {
+		// The event source may be a federated resource that was
+		// deleted, but is more likely a non-federated resource in the
+		// target cluster.
+		glog.V(7).Infof("%s %q was not found which indicates that the %s is not federated",
```

Comment:
[Query]: I doubt that this would happen. The `federatedController` is started only when this type is federated eventually resulting in objects in `federatedStore`. The resulting action if this type in no longer federated is stopping the sync controller, which means that this `reconcile` loop using this `accessor` is already stopped. 
Even if the sync controller `reconcile` loop waits to clear all the residual events, the typeconfig controller wouldn't have completed the  `disable/delete` (or `unfederate`) operation.
It does not harm to keep this log info any which ways; it would be great if you can correct/clear my understanding. Is this meant only for namespaces?

Reply:
For a given federated type, reconciliation is triggered on changes for all target resources in member clusters.  It is very likely that some of those target resources are local to a member cluster and have no equivalent federated resource in the host cluster (e.g. the deployments managing system components in the `kube-system` namespace).

Reply:
Thanks for the clarification.